index.js

```diff
@@ -79,8 +86,27 @@ Object.defineProperty(SQLStatement.prototype, 'sql', {
  * @param {...any} values
  * @returns {SQLStatement}
  */
-function SQL(strings) {
-  return new SQLStatement(strings.slice(0), Array.from(arguments).slice(1))
+function SQL(tpl, ...val) {
+  const strings = [tpl[0]]
+  const values = []
+  for (let { length } = tpl, prev = tpl[0], j = 0, i = 1; i < length; i++) {
+    const current = tpl[i]
+    const value = val[i - 1]
+    if (/^('|")/.test(current) && RegExp.$1 === prev.slice(-1)) {
+      strings[j] = [strings[j].slice(0, -1), value, current.slice(1)].join('`')
```

Comment:
This doesn't seem to escape the value (i.e. escape backticks inside the string). I also don't think it's easily possible because different dialects have different special chars.

Reply:
"_I challenge you_" finding anywhere someone using a name with a backtick in it for a table or a field name.

The being said, how about we make them configurable? backtick by default, `setEscape('"')` to change it as double quotes.

The escaping of the symbol is unnecessary at this point, nobody will have issues since this is a new feature, and if they want to use back ticks for table or field names, they are in charge of passing these already escaped.

YAGNI

Reply:
It's not about having table names with backticks or not, it's about security. If people expect this to escape the value then they might pass strings derived from user input. If it's not escaped properly, that would allow SQL injection.

Reply:
OK, current version exposes a method to define quotes: `SQL.withQuote('`')`

I could use that and fallback with a default based on back-ticks, or, if no quote is defined, simply throw an error.

I can escape automatically within the quotes, as long as escaping works with backslashes in every SQL implementation.

Is that the case?

Reply:
current version has a test too, with auto escape and error on failure

Reply:
off topic: I think `SQL.withQuotes` would feel more natural to devs ... since I usually refer to quotes in plural 🤔

Reply:
`SQL.withQuotes('`')` it is 😅